Service/Push/OrderRequestService.php

```diff
@@ -0,0 +1,327 @@
+     * @return Order|OrderPayment
+     * @throws \Exception
+     */
+    public function getOrderByRequest(?PushRequestInterface $pushRequest = null): Order|OrderPayment
+    {
+        if ($this->order->getId()) {
+            return $this->order;
+        } else {
+            $brqOrderId = $this->getOrderIncrementIdFromRequest($pushRequest);
+
+            $this->order->loadByIncrementId((string)$brqOrderId);
+
+            if (!$this->order->getId()) {
+                $this->logging->addDebug('Order could not be loaded by Invoice Number or Order Number');
+                // Try to get order by transaction id on payment.
+                $this->order = $this->getOrderByTransactionKey($pushRequest);
+            }
+        }
+
+        return $this->order;
+    }
+
+    /**
+     * Get the order increment ID based on the invoice number or order number.
+     *
+     * @param $pushRequest
+     * @return string|null
+     */
+    protected function getOrderIncrementIdFromRequest($pushRequest): ?string
+    {
+        $brqOrderId = null;
+
+        if (!empty($pushRequest->getInvoiceNumber()) && strlen($pushRequest->getInvoiceNumber()) > 0) {
+            $brqOrderId = $pushRequest->getInvoiceNumber();
+        }
+
+        if (!empty($pushRequest->getOrderNumber()) && strlen($pushRequest->getOrderNumber()) > 0) {
+            $brqOrderId = $pushRequest->getOrderNumber();
+        }
+
+        return $brqOrderId;
+    }
+
+    /**
+     * Sometimes the push does not contain the order id, when that's the case try to get the order by his payment,
+     * by using its own transaction key.
+     *
+     * @param $pushRequest
+     * @return OrderPayment
+     * @throws \Exception
+     */
+    protected function getOrderByTransactionKey($pushRequest): OrderPayment
+    {
+        $trxId = $this->getTransactionKey($pushRequest);
+
+        $this->transaction->load($trxId, 'txn_id');
+        $order = $this->transaction->getOrder();
+
+        if (!$order) {
+            throw new BuckarooException(__('There was no order found by transaction Id'));
+        }
+
+        return $order;
+    }
+
+    /**
+     * Retrieves the transaction key from the push request.
+     *
+     * @param $pushRequest
+     * @return string
+     */
+    protected function getTransactionKey($pushRequest): string
+    {
+        $trxId = '';
+
+        if (!empty($pushRequest->getTransactions())) {
+            $trxId = $pushRequest->getTransactions();
+        }
+
+        if (!empty($pushRequest->getDatarequest())) {
+            $trxId = $pushRequest->getDatarequest();
+        }
+
+        if (!empty($pushRequest->getRelatedtransactionRefund())) {
+            $trxId = $pushRequest->getRelatedtransactionRefund();
+        }
+
+        return $trxId;
+    }
+
+    /**
+     * Try to add a notification note to the order comments.
+     *
+     * @param Phrase|string $message
+     */
+    public function setOrderNotificationNote(Phrase|string $message): void
```

Comment:
where is the message variable used in?

Reply:
Here ->
$note = 'Buckaroo attempted to update this order, but failed: ' . $message;